src/components/Dialogs/BulkRenameFilesDialog.vue

```diff
@@ -0,0 +1,317 @@
+      }
+      item.notChanged = item.name === item.targetName
+    }
+  })
+  hasDuplicated.value = false
+  const allTarget = new Map()
+  items
+    .filter(item => !!item.targetFullName)
+    .forEach(item => {
+      allTarget.set(item.targetFullName, (allTarget.get(item.targetFullName) || 0) + 1)
+    })
+  items.forEach(item => {
+    item.duplicated = allTarget.get(item.targetFullName) > 1
+    if (item.duplicated) {
+      hasDuplicated.value = true
+    }
+  })
+}
+
+const run = async () => {
+  if (!candidateItems.value.length) {
+    return toast.warn(t('dialogs.bulkRenameFiles.nothing_no_do'))
+  }
+  const reqList = []
+  for (const item of candidateItems.value) {
+    reqList.push(contentStore.renameTorrentFile(props.hash, item.fullName, item.targetFullName!))
+  }
+  running.value = true
+  Promise.all(reqList)
+    .then(() => {
+      toast.success(t('dialogs.bulkRenameFiles.success'))
+    })
+    .catch(e => {
+      toast.error(e.toString())
+    })
+    .finally(() => {
+      running.value = false
+      contentStore.updateFileTree()
+      // close dialog, because haven't a good way to refresh torrent files after rename
+      // because bulkRenameFilesDialog can be create with subfolder
+      close()
+    })
+}
+
+const close = () => {
+  isOpened.value = false
+}
+
+watch([regexpInput, regexpFlagsInput, targetInput], () => {
+  dryRunRename()
+})
+
+onMounted(() => {
+  parseNode(props.node)
+})
+</script>
+
+<template>
+  <v-dialog v-model="isOpened" persistent>
+    <v-card density="compact">
+      <v-card-title>
+        {{ $t('dialogs.bulkRenameFiles.title') }}
+        <v-btn style="float: right" icon="mdi-close" variant="plain" density="compact" @click="close()"></v-btn>
+      </v-card-title>
+      <v-card-text>
+        <v-form v-model="isFormValid" ref="form">
+          <v-text-field hide-details density="compact" v-model="regexpInput" :rules="rules" :label="$t('dialogs.bulkRenameFiles.regexp')">
+            <template v-slot:append>
+              <v-select
+                class="flags-select"
+                v-model="regexpFlagsInput"
+                :items="['d', 'g', 'i', 'm', 's', 'u', 'v', 'y']"
+                hint="Select Regular Expression Flags"
+                label="Flags"
+                density="compact"
+                multiple
+                hide-details></v-select>
+            </template>
+          </v-text-field>
+          ->
+          <v-text-field hide-details density="compact" v-model="targetInput" :rules="rules" :label="$t('dialogs.bulkRenameFiles.target')" />
+          <v-badge color="success" :content="candidateItems.length">
+            <v-btn :loading="running" :disabled="!isFormValid || hasDuplicated" color="primary" @click="run()">{{ $t('dialogs.bulkRenameFiles.run') }}</v-btn>
+          </v-badge>
+        </v-form>
+        <v-data-table-virtual :headers="headers" :items="items" density="compact" height="calc(100vh - 200px)" fixed-header>
```

Comment:
UI not responsive at all, please rework by using virtual scroller instead of virtual data table.

See Content view for example.

Reply:
I haven't found a good solution. Vertical display might be tricky to handle. Currently, I have it set to not wrap, allowing horizontal scrolling.

![image](https://github.com/VueTorrent/VueTorrent/assets/45785585/d513a798-1cd6-4627-9701-1382d083ef12)

Reply:
Can you try putting the diff on 2 lines instead of next to each other?

I think of putting files without change one line like the current layout but changing files would display on two lines, with the top one being the "old" one, highlighted in red or striked-through. And the "new" one being at the bottom and green?

Reply:
I don't know which one is better,imo I like one line display😂, two lines in one <td> maybe too compact.

Reply:
That's why I suggested using virtual scroller instead of using the data table 😄

Reply:
What about this?

https://github.com/VueTorrent/VueTorrent/assets/45785585/2ba72c12-1a6b-439b-92f5-8167c77c4742